src/commands/domain.js

```diff
@@ -302,3 +312,118 @@ function sortDomains (parsedDomains) {
 function printlnWithIndent (text, indentLevel) {
   Logger.println(' '.repeat(indentLevel) + text);
 }
+
+export async function overview (params) {
+
+  const { format, filter } = params.options;
+
+  const summary = await getSummary().then(sendToApi);
+  const consoleUrl = summary.user.partnerConsoleUrl;
+
+  const applications = [
+    ...summary.user.applications.map((app) => {
+      return { ownerName: summary.user.name, ownerId: summary.user.id, ...app };
+    }),
+    ...summary.organisations.flatMap((o) => {
+      return o.applications.map((app) => {
+        return { ownerName: o.name, ownerId: o.id, ...app };
+      });
+    }),
+  ];
+
+  const applicationsWithDomains = await Promise.all(
+    applications.map(async (app) => {
+      const domains = await getAllDomains({ id: app.ownerId, appId: app.id }).then(sendToApi);
+      return { app, domains };
+    }),
+  );
+
+  const applicationsWithParsedDomain = applicationsWithDomains
+    .flatMap(({ app, domains }) => {
+      return domains
+        .filter((domain) => filter == null || domain.fqdn.includes(filter))
+        .map((domain) => {
+
+          const parsedDomain = parseDomain(domain.fqdn);
+          const pathname = new URL('https://' + domain.fqdn).pathname;
+          const subdomains = parsedDomain.subdomain !== '' ? parsedDomain.subdomain.split('.') : [];
+
+          // We're trying to create an propertyPath for lodash to create a tree structure object,
+          // the propertyPath for `aaa.bbb.ccc.example.com/the-path` would be:
+          // ["example.com", "example.com.ccc", "example.com.ccc.bbb", "example.com.ccc.bbb.aaa", "/path-aaa"]",
+
+          const sortSegments = [parsedDomain.domain, ...subdomains.reverse()];
+          const propetyPath = sortSegments.map((item, i, all) => {
+            return all.slice(0, i + 1).reverse().join('.');
+          });
+          propetyPath.push(pathname);
+
+          return {
+            ownerId: app.ownerId,
+            ownerName: app.ownerName,
+            appId: app.id,
+            appName: app.name,
+            appConsoleUrl: `${consoleUrl}/${app.id}`,
+            appVariantSlug: app.variantSlug,
+            domain: domain.fqdn,
+            propetyPath,
+          };
+        });
+    });
+
+  const applicationsWithParsedDomainAsTree = {};
+  for (const { propetyPath, ...appWithDomain } of applicationsWithParsedDomain) {
+    _.set(applicationsWithParsedDomainAsTree, propetyPath, appWithDomain);
+  }
+
+  const applicationsWithParsedDomainAsSortedTree = recursiveSort(applicationsWithParsedDomainAsTree);
+
+  switch (format) {
+    case 'json':
+      Logger.printJson(applicationsWithParsedDomainAsSortedTree);
+      break;
+    case 'human':
+    default:
+      if (Object.keys(applicationsWithParsedDomainAsSortedTree).length === 0) {
+        Logger.println(`No matches for filter "${filter}"`);
+      }
```

Comment:
**nitpick:** when you have no domain at all, the result of the command is `No matches for filter ""`

We could split into 2 different cases (no result + filter active vs no result + filter inactive)?

Reply:
Good idea!